### PR TITLE
Parallel paco links

### PR DESCRIPTION
**Pull request by efcaguab**
Added parallel evaluation capabilities to `paco_links()` using `foreach()`. It spawn an independent process for each being jackknifed . If evaluating the contributions sequentially, it can also add a progress bar (borrowed from `plyr::create_progress_bar ()`). Updated corresponding roxygen comments.

### CODE REVIEW

File: R/paco_links.r [hunk 1]
```diff
@@ -1,24 +1,35 @@
...
-   } 
+   nlinks <- sum(D$HP)
+   
+   # In parallel
+   if (.parallel & exists ("foreach")) {
+     foreach (i=1:nlinks, .combine = rbind) %dopar% single_paco_link (D, HP.ones, i, ...)
+   } else {
+   # In sequence 
+     if (.parallel & !exists ("foreach")) warning ("No parallel backend registered. Executing sequentially instead.")
+     pb <- plyr::create_progress_bar (.progress)
```
Comment (tpoisot) [inline R/paco_links.r:22]: This _really_ should be optional. Printing things to any output in R is decreasing performnance a lot.

**Comment by tpoisot** [comment]
If possible, please start from the head branch and not from a fork. Luckily this time they were in sync.

File: R/paco_links.r [hunk 2]
```diff
@@ -1,24 +1,35 @@
 #' Contribution of individual links
 #' @param D A list returned by proc_analysis
+#' @param .parallel if \code{TRUE}, calculate the jacknife contribution in parallel using the backend provided by foreach
+#' @param .progress name of the progress bar to use see \code{\link[plyr]{create_progress_bar}}. Options inlcude "text" and "tk". It only works when \code{parallel = FALSE}
```
Comment (tpoisot) [inline R/paco_links.r]: There is a typo (`inlcude` instead of `include`)

**Comment by tpoisot** [comment]
I've added two comments for two lines -- once this is sorted out, I'll merge, thanks for that.

Any reason you picked `foreach` instead of `doMC` for the parallel backend?

File: R/paco_links.r (hunk 1 above)
Comment (efcaguab) [inline R/paco_links.r:22]: It is optional. The argument defaults to `.progress = "none"`, and when `create_progress_bar` is called with the default argument it returns a set of empty functions—so nothing is being printed in the console. 

```
> plyr::create_progress_bar ("none")
$init
function (x) 
{
}
<environment: 0x12d676638>

$step
function () 
{
}
<environment: 0x12d676638>

$term
function () 
{
}
```

File: R/paco_links.r (hunk 2 above)
Comment (efcaguab) [inline R/paco_links.r]: Good spot! Thanks

**Comment by efcaguab** [comment]
Thanks Tim. I've corrected the typo you found. Good spot! Also, sorry for not starting from the head, as you can clearly see I'm still quite unexperienced using git... 

Regarding `foreach`: I might be wrong, but I think it provides the looping construct, while `doMC` is just a backend for it (just like `doParallel` and `doSNOW`). I myself use `doMC`, but other users might want/need to use other backend.

File: R/paco_links.r (hunk 1 above)
Comment (tpoisot) [inline R/paco_links.r:22]: OK! I was not sure how this works.

**Comment by tpoisot** [comment]
I'm _almost_ ready to merge -- please add your name + email in the `DESCRIPTION` file, as Matt did. Once this is done, we'll be good to go.

**Comment by tpoisot** [comment]
As soon as #4 is sorted out, I will merge these changes. Good job!

**Comment by efcaguab** [comment]
Why does #4 need to be sorted out before merging?

**Comment by tpoisot** [comment]
Yeah, parallel! :+1:

**Pull request merged**